### ISSUE TITLE
`sensitivityTimeProfiles` performance

### ISSUE THREAD

**Issue by PavelBal** [esqLABS/esqlabsR#204]
Calling `sensitivityTimeProfiles` is slow, have to profile and try to optimize the performance.

**Comment by IndrajeetPatil** [comment]
I am not sure how I can optimize this further. 

The time taken is going to depend on how many output paths, parameter paths, PK parameters, and variations have been specified. 

For example, for a **single** output and parameter path, and only three parameter factors and PK parameters:

Median time taken to prepare and print the plot: 200 ms
The number of data points plotted: 9329

``` r
library(esqlabsR)

simPath <- system.file("extdata", "Aciclovir.pkml", package = "ospsuite")
simulation <- loadSimulation(simPath)
outputPaths <- "Organism|PeripheralVenousBlood|Aciclovir|Plasma (Peripheral Venous Blood)"
parameterPaths <- "Aciclovir|Lipophilicity"

ls_results <- sensitivityCalculation(
  simulation = simulation,
  outputPaths = outputPaths,
  parameterPaths = parameterPaths,
  variationRange = c(1, 5, 10)
)

bench::mark(
  sensitivityTimeProfiles(ls_results),
  check = FALSE
)
#> Warning: Some expressions had a GC in every iteration; so filtering is disabled.
#> # A tibble: 1 x 6
#>   expression                             min median `itr/sec` mem_alloc `gc/sec`
#>   <bch:expr>                          <bch:> <bch:>     <dbl> <bch:byt>    <dbl>
#> 1 sensitivityTimeProfiles(ls_results)  187ms  194ms      5.01    7.19MB     5.01
```

<sup>Created on 2022-04-08 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1.9000)</sup>

But with three parameter paths and 19 parameter factors:

Median time taken to prepare and print the plot: 3000 ms
The number of data points plotted: 27987

``` r
library(esqlabsR)

simPath <- system.file("extdata", "Aciclovir.pkml", package = "ospsuite")
simulation <- loadSimulation(simPath)
outputPaths <- "Organism|PeripheralVenousBlood|Aciclovir|Plasma (Peripheral Venous Blood)"
parameterPaths <- c(
  "Aciclovir|Lipophilicity",
  "Applications|IV 250mg 10min|Application_1|ProtocolSchemaItem|Dose",
  "Neighborhoods|Kidney_pls_Kidney_ur|Aciclovir|Glomerular Filtration-GFR|GFR fraction"
)

ls_results <- sensitivityCalculation(
  simulation = simulation,
  outputPaths = outputPaths,
  parameterPaths = parameterPaths
)

bench::mark(
  sensitivityTimeProfiles(ls_results),
  check = FALSE
)
#> Warning: Some expressions had a GC in every iteration; so filtering is disabled.
#> # A tibble: 1 x 6
#>   expression                             min median `itr/sec` mem_alloc `gc/sec`
#>   <bch:expr>                          <bch:> <bch:>     <dbl> <bch:byt>    <dbl>
#> 1 sensitivityTimeProfiles(ls_results)  3.35s  3.35s     0.299    43.9MB     5.38
```

<sup>Created on 2022-04-08 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1.9000)</sup>

And, although, one might naively expect that if the number of data points is tripled, the time should only triple, the problem is that with multiple parameter paths, we are creating small multiple plots. 
In other words, we are now creating and printing not a single but multiple plots and arranging them, which takes even more time.

**Comment by IndrajeetPatil** [comment]
Oh, also, keep in mind that we are not providing this function a data frame, like we do in `sensitivitySpiderPlot()`.

We first need to extract data frames internally using `.simulationResultsBatchToTimeSeriesDataFrame()`. This is a consequence of https://github.com/esqLABS/esqlabsR/issues/153.

**Comment by IndrajeetPatil** [comment]
So you can't expect it to be as fast as `sensitivitySpiderPlot()`, if that's indeed your point of comparison.

**Comment by PavelBal** [comment]
Still an issue...
